frontend/src/v4/modules/viewpoints/viewpoints.selectors.ts

```diff
@@ -35,25 +37,49 @@ export const groupsOfViewpoint = function*(viewpoint) {
 			if (Array.isArray(viewpoint[prop])) { // if the property is an array of groupId
 				const groupsIds: string[] = viewpoint[prop];
 				for (let j = 0; j < groupsIds.length; j++ ) {
-					yield groupsIds[j];
+					ids.add(groupsIds[j]);
 				}
 			} else {// if the property is just a groupId
-				yield viewpoint[prop];
+				ids.add(viewpoint[prop]);
 			}
 		}
 	}
+
+	return ids;
 };
 
-export const isViewpointLoaded = (viewpoint, groups) => {
-	let areGroupsLoaded = true;
+export const getLoadedGroupsIds = function (viewpoint) {
```

Comment:
This is essentially the same code as `getGroupsIDsOfViewpoint`, with the sole difference that here you map everything to the ids, and the `groupsProperties` are different.

Some refactoring could be used here